### PR TITLE
add checksums workflow

### PR DESCRIPTION
**Pull request by ryanfortner**
This workflow will run on each commit to the main branch.

### CODE REVIEW

**Comment by ryanfortner** [comment]
@Itai-Nelken

**Comment by Itai-Nelken** [comment]
Sorry for not answering, I forgot about this pr.
could you use a action to commit? for example [this one](https://github.com/marketplace/actions/add-commit).

**Comment by ryanfortner** [comment]
> Sorry for not answering, I forgot about this pr. could you use a action to commit? for example [this one](https://github.com/marketplace/actions/add-commit).

Why do we need to?

File: .github/workflows/checksums.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  generate-checksums:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - uses: actions/checkout@v2
+      - name: Create checksums
+        run: |
+          git config user.name 'GitHub Actions'
```
Comment (Itai-Nelken) [inline .github/workflows/checksums.yml:22]: Are those username and email valid/work with git (have access to the)?

Comment (ryanfortner) [reply .github/workflows/checksums.yml:22]: I believe so. But I think we should put the user name/email of a user with write access.

Comment (ryanfortner) [reply .github/workflows/checksums.yml]: @Itai-Nelken I've put Botspot as the committer. It should work now.

Comment (Itai-Nelken) [reply .github/workflows/checksums.yml]: @Botspot is this fine with you?

**Comment by Botspot** [comment]
What is this and why do we need it?

**Comment by ryanfortner** [comment]
> What is this and why do we need it?

This is a GitHub Actions workflow, used to generate the sha256 sums of each file in this repo and upload them in a single text file. 

It's not completely necessary, but it can be used to verify the integrity of files hosted here.

**Comment by Botspot** [comment]
> > What is this and why do we need it?
> 
> This is a GitHub Actions workflow, used to generate the sha256 sums of each file in this repo and upload them in a single text file.
> 
> It's not completely necessary, but it can be used to verify the integrity of files hosted here.

Sounds reasonable. This PR is fine with me.

**Comment by Itai-Nelken** [comment]
Ready to merge @ryanfortner?

**Comment by ryanfortner** [comment]
> Ready to merge @ryanfortner? 

Yes.

**Comment by Itai-Nelken** [comment]
@ryanfortner the action checksums the sha256sums.txt` file, and it does it before writing it so it's incorrect. I made the action remove it before generating the new sums.

**Pull request merged**